src/core/state.h

```diff
@@ -65,15 +65,39 @@ class State {
      * Constructor.
      * @param db Pointer to the database.
      * @param storage Pointer to the blockchain's storage.
-     * @param rdpos Pointer to the rdPoS object.
      * @param p2pManager Pointer to the P2P connection manager.
      * @param options Pointer to the options singleton.
      * @throw DynamicException on any database size mismatch.
      */
-    State(DB& db, Storage& storage, rdPoS& rdpos, P2P::ManagerNormal& p2pManager, const Options& options);
+    State(DB& db, Storage& storage, P2P::ManagerNormal& p2pManager, const Options& options);
 
     ~State(); ///< Destructor.
 
+    // ======================================================================
+    // RDPOS WRAPPER FUNCTIONS
+    // ======================================================================
+
+    ///@{
+    /** Wrapper for the respective rdPoS function. */
```

Comment:
What about a single getRdPoS() / getConsensus() method instead that returns a rdPoS& ?

Reply:
There was one originally, but it was removed because @itamarcps told me it would be better to not expose rdPoS at all to other classes, it should be completely isolated and State should handle things instead so they don't risk de-syncing with each other due to someone else's accessing it.